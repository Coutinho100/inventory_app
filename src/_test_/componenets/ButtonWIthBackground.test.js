import React from 'react';
import ButtonWIthBackground from '../../components/Component/ButtonWIthBackground';

import renderer from 'react-test-renderer';
import { View, Text, TouchableNativeFeedback, Platform } from 'react-native';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { render, cleanup, fireEvent } from 'react-native-testing-library';
import test_function from '../componenets/test_function';

configure({ adapter: new Adapter() });

const onPress = jest.fn();
const baseProps = {
    onPress
}
test_function(["<text render/>",1],['renders correctly','renders the length of view'],[<ButtonWIthBackground />, View])

describe("< button/>", () => {
    beforeEach(() => {
        props = {
            Platform: "android"
        }
    });
   
    it('check  view text', () => {
        const wrapper = shallow(<ButtonWIthBackground />);
        wrapper.setProps({ disabled: true });
        // console.log(wrapper.debug());
        expect(wrapper.find(View).length).toBe(1)
        expect(wrapper.find(Text).length).toBe(1)
        // expect(wrapper.find(TouchableNativeFeedback)).toEqual(1)
    });
    // it("onPress check", () => {
    //     const wrapper = shallow(<ButtonWIthBackground {...baseProps} />);
    //     wrapper.simulate("onPress");
    //     // expect(baseProps.onPress).toHaveBeenCalled();
    // });

    it('onPress check', () => {
        const rendered = render(<ButtonWIthBackground {...baseProps} />);
        const Component = rendered.getByTestId('messageText');
        fireEvent(Component, 'press');

    });
});






