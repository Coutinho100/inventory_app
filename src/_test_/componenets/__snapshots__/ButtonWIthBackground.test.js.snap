// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< button/> renders correctly 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="messageText"
>
  <View
    style={
      Array [
        Object {
          "borderColor": "black",
          "borderRadius": 5,
          "borderWidth": 1,
          "margin": 5,
          "padding": 10,
        },
        Object {
          "backgroundColor": undefined,
        },
        null,
      ]
    }
  >
    <Text
      style={null}
    />
  </View>
</View>
`;
